vlrouter.go

```diff
@@ -769,6 +817,32 @@ func (self *Vlrouter) initFgraph() error {
 	})
 	bcastSrcFlow.Next(sw.DropAction())
 
+	// redirect dns requests to controller
+	macSaMask := net.HardwareAddr{0xFF, 0xFF, 0x00, 0x00, 0x00, 0x00}
+	macSa := net.HardwareAddr{0x02, 0x02, 0x00, 0x00, 0x00, 0x00}
+	dnsFlow1, _ := self.inputTable.NewFlow(ofctrl.FlowMatch{
+		Priority:   DNS_FLOW_MATCH_PRIORITY,
+		MacSa:      &macSa,
+		MacSaMask:  &macSaMask,
+		Ethertype:  protocol.IPv4_MSG,
+		IpProto:    protocol.Type_UDP,
+		UdpDstPort: 53,
+	})
+	dnsFlow1.Next(sw.SendToController())
+
+	// re-inject dns requests
+	dnsFlow2, _ := self.inputTable.NewFlow(ofctrl.FlowMatch{
```

Comment:
Same as above. minor nit: rename dnsFlow2 to a semantically meaningful one (suggestion: dnsReinjectFlow)